src/api/fetch.js

```diff
@@ -35,101 +35,86 @@ export async function* indexGenerator(path, auth) {
   }
 }
 
+async function resolveRequest(request) {
+  const response = await fetch(request).catch((reason) => {
+    throw { error: [reason] };
+  });
+  const result = await response.json();
+  if (response.ok) {
+    return result;
+  } else {
+    throw result;
+  }
+}
+
 export function create(path, auth, object) {
```

Comment:
All of these functions should be marked as async and `return await resolveRequest(...)`, IMO. Now, looking at just the function, you wouldn't know that it returns asynchronously.